File: psclient/src/main/java/com/majeur/psclient/util/Preferences.java
package com.majeur.psclient.util;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class Preferences {

    private static SharedPreferences preferences(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    public static String getStrPreference(Context context, String key) {
        return preferences(context).getString(key, null);
    }

    public static boolean getBoolPreference(Context context, String key) {
        return preferences(context).getBoolean(key, false);
    }

    public static void setPreference(Context context, String key, String value) {
        preferences(context).edit().putString(key, value).apply();
    }

    public static void setPreference(Context context, String key, boolean value) {
        preferences(context).edit().putBoolean(key, value).apply();
    }
}

File: psclient/src/main/java/com/majeur/psclient/util/Preferences.kt
package com.majeur.psclient.util

import android.content.Context

object Preferences {

    private const val PREF_NAME = "user-preferences"
    private const val KEY_NEWS_BANNER = "news-banner-enabled"
    private const val KEY_BATTLE_SOUND = "battle-sound-enabled"

    fun isNewsBannerEnabled(c: Context) = readBool(c, KEY_NEWS_BANNER, true)
    fun setNewsBannerEnabled(c: Context, value: Boolean) = writeBool(c, KEY_NEWS_BANNER, value)

    fun isBattleSoundEnabled(c: Context) = readBool(c, KEY_BATTLE_SOUND, true)
    fun setBattleSoundEnabled(c: Context, value: Boolean) = writeBool(c, KEY_BATTLE_SOUND, value)

    private fun readBool(c: Context, key: String, def: Boolean) = get(c).getBoolean(key, def)
    private fun writeBool(c: Context, key: String, value: Boolean) = get(c).edit().putBoolean(key, value).apply()

    private fun get(c: Context) = c.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)

}